x-pack/plugins/task_manager/server/queries/mark_available_tasks_as_claimed.ts

```diff
@@ -109,3 +109,25 @@ export const updateFields = (fieldUpdates: {
   lang: 'painless',
   params: fieldUpdates,
 });
+
+export const updateFieldsAndMarkAsFailed = (
+  fieldUpdates: {
+    [field: string]: string | number | Date;
+  },
+  taskMaxAttempts: { [field: string]: number }
+): ScriptClause => ({
+  source: `
+  if (params.taskMaxAttempts[ctx._source.task.taskType] == null || (params.taskMaxAttempts[ctx._source.task.taskType] != null && ctx._source.task.attempts < params.taskMaxAttempts[ctx._source.task.taskType])) {
+    ctx._source.task.status = "claiming"; ${Object.keys(fieldUpdates)
+      .map((field) => `ctx._source.task.${field}=params.fieldUpdates.${field};`)
+      .join(' ')}
+  } else {
+    ctx._source.task.status = "failed";
+  }
+  `,
+  lang: 'painless',
+  params: {
+    fieldUpdates,
+    taskMaxAttempts,
+  },
+});
```

Comment:
This _looks_ right, but obviously hard to be sure.
I know this is still in draft, so just assuming you haven' got around to it yet, but lets make sure we have clear unit tests for this as they can act as good documentation of what this should end up evaluating as and make it easier to maintain.

Reply:
Unit tests have been updated to reflect what the script should look like.